controllers/ndbserver_controller_helpers.go

```diff
@@ -0,0 +1,84 @@
+package controllers
+
+import (
+	"context"
+
+	ndbv1alpha1 "github.com/nutanix-cloud-native/ndb-operator/api/v1alpha1"
+	"github.com/nutanix-cloud-native/ndb-operator/common"
+	"github.com/nutanix-cloud-native/ndb-operator/common/util"
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_api"
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Fetches all databases from the NDB API and converts them to
+// NDBServerDatabaseInfo type object (to be consumed by the NDBServer CR)
+func getNDBServerDatabasesInfo(ctx context.Context, ndbClient *ndb_client.NDBClient) (databases []ndbv1alpha1.NDBServerDatabaseInfo, err error) {
+	log := log.FromContext(ctx)
+	log.Info("Fetching and converting databases from NDB")
+
+	dbs, err := ndb_api.GetAllDatabases(ctx, ndbClient)
+	if err != nil {
+		log.Error(err, "NDB API error while fetching databases")
+		return
+	}
+	databases = make([]ndbv1alpha1.NDBServerDatabaseInfo, len(dbs))
+	for i, db := range dbs {
+		databaseInfo := ndbv1alpha1.NDBServerDatabaseInfo{
+			Name:   db.Name,
+			Id:     db.Id,
+			Status: db.Status,
+		}
+		if len(db.DatabaseNodes) > 0 {
+			databaseInfo.DBServerId = db.DatabaseNodes[0].DatabaseServerId
+			if len(db.DatabaseNodes[0].DbServer.IPAddresses) > 0 {
```

Comment:
Will you always pick 0th entry "db.DatabaseNodes[0]"? Just making sure.

Reply:
Yes, for single instance databases we will always pick the 0th entry.

Reply:
Makes sense, please add a comment there for better context.